File: ltropik-client/src/api/schedule.ts
import api from './client';
import type { ScheduleEntry } from '../types';

export const getSchedule = (from?: string, to?: string) =>
  api.get<ScheduleEntry[]>('/schedule', { params: { from, to } });

export const createSchedule = (data: {
  lessonId: string;
  startsAt: string;
  durationMinutes: number;
  notes?: string;
  teacherId?: string;
}) => api.post('/schedule', data);

export const updateSchedule = (id: string, data: { startsAt?: string; durationMinutes?: number; notes?: string; teacherId?: string }) =>
  api.put(`/schedule/${id}`, data);

export const deleteSchedule = (id: string) => api.delete(`/schedule/${id}`);

File: src/LTropik.Domain/Entities/Schedule.cs
namespace LTropik.Domain.Entities;

public class Schedule
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid LessonId { get; set; }
    public Guid TeacherId { get; set; }
    public DateTimeOffset StartsAt { get; set; }
    public int DurationMinutes { get; set; } = 60;
    public string? Notes { get; set; }

    public Lesson Lesson { get; set; } = null!;
    public User Teacher { get; set; } = null!;
}
